src/sentry/static/sentry/app/actionCreators/environments.jsx

```diff
@@ -1,8 +1,18 @@
+import OrganizationEnvironmentsStore from 'app/stores/organizationEnvironmentsStore';
+
 /**
  * Fetches all environments for an organization
  *
  * @param {String} organizationSlug The organization slug
  */
-export function fetchOrganizationEnvironments(api, organizationSlug) {
-  return api.requestPromise(`/organizations/${organizationSlug}/environments/`);
+export async function fetchOrganizationEnvironments(api, organizationSlug) {
+  let environments;
+  try {
+    environments = await api.requestPromise(
+      `/organizations/${organizationSlug}/environments/`
+    );
+    OrganizationEnvironmentsStore.loadInitialData(environments);
```

Comment:
not sure if these (`loadInitialData`, `onFetchOrganization...`) should be converted to actions instead of interacting with the store directly

Reply:
It's better if they are actions, that way you could have other stores listen to those actions.